src/playground/sample_data_tooling/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,336 @@
+        timestamp = timestamp,
+        minutes = minutes,
+        current_date = current_date,
+        max_bulk_size = max_bulk_size,
+        anomaly_detection_trend = anomaly_detection_trend
+    )
+
+    dataset = []
+
+    # If the user provides their own data
+    if file_provided:
+        dataset = ingest_from_user_data(filename = data_template)
+
+    # If anomalies wanted to be generated
+    elif anomaly_detection_trend:
+        for i in range(number):
+            # Generate random data
+            entry = generate_data(data_template, mapping)
+            
+            # For each feature needing a trend, modify that specific field value to fit a trend
+            for desired_trend in anomaly_detection_trend:
+                if desired_trend["data_trend"] == "AverageTrend":
+                    changed_entry = AverageTrend(  
+                        timestamp = timestamp, 
+                        entry = entry,
+                        feature_trend = desired_trend, 
+                        current_date = current_date
+                    )
+                    entry = changed_entry.generate_data_trend()
+            dataset.append(entry)
+            # Increment current date
+            current_date += timedelta(minutes = minutes)
+    else:
+         # Generates the specified number of documents
+        for i in range(number):
+            entry = generate_data(data_template, mapping)
+            if type(entry) is list:
+                for element in entry:
+                    if type(element) is str:
+                        element = loads(element)
+                    if timestamp:
+                        element[timestamp] = current_date
+                    element = dumps(element)
+                    dataset.append(element)
+            else:
+                entry = generate_data(data_template, mapping)
+                if type(entry) is str:
+                    entry = loads(entry)
+                if timestamp:
+                    entry[timestamp] = current_date
+                entry = dumps(entry)
+                dataset.append(entry)
+            current_date += timedelta(minutes = minutes)
+
+    # Calls BULK API to ingest documents of size "chunk"
+    i = 0
+    while i < len(dataset):
+
+        # Build request body
+        request_building = build_request_body(index_name, file_provided, timestamp, minutes, chunk, current_index = i, dataset = dataset, max_bulk_size = max_bulk_size)
+        request_body = request_building[0]
+        i = request_building[1]
+
+        # Validates that request_body is both a list and that it is of even length
+        if type(request_body) is not list:
+            raise TypeError("Request body is not a list")
+        if len(request_body) % 2 != 0:
+            raise ValueError("Request body does not come in index, action pairs; an index name does not have an action or vice versa") 
+            
+        response = client.bulk(body = request_body)
```

Comment:
do we need to try catch for the client bulk call ?

Reply:
Yes, I have added a try catch to all API calls.